fe/fe-core/src/main/java/com/starrocks/catalog/ExpressionRangePartitionInfo.java

```diff
@@ -35,21 +49,31 @@
 public class ExpressionRangePartitionInfo extends RangePartitionInfo {
 
     @SerializedName(value = "partitionExprs")
-    private final List<Expr> partitionExprs;
+    private List<Expr> partitionExprs;
+
+    public ExpressionRangePartitionInfo() {
+        this.type = PartitionType.EXPR_RANGE;
+    }
+
 
-    public ExpressionRangePartitionInfo(List<Expr> partitionExprs, List<Column> columns) {
+    public ExpressionRangePartitionInfo(List<Expr> partitionExprs, List<Column> columns, PartitionType type) {
         super(columns);
         Preconditions.checkState(partitionExprs != null);
         Preconditions.checkState(partitionExprs.size() > 0);
         Preconditions.checkState(partitionExprs.size() == columns.size());
         this.partitionExprs = partitionExprs;
         this.isMultiColumnPartition = partitionExprs.size() > 0;
+        this.type = type;
```

Comment:
this is ambiguous. the default type is EXPR_RANGE. add some comment to explain why can change type?